recipes/rmlui/3.3/conanfile.py

```diff
@@ -0,0 +1,132 @@
+from conans.model.version import Version
+from six import StringIO
+import os
+
+
+class RmluiConan(ConanFile):
+    name = "rmlui"
+    description = "RmlUi - The HTML/CSS User Interface Library Evolved"
+    homepage = "https://github.com/mikke89/RmlUi"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    topics = ("conan", "css", "gui", "html", "lua", "rmlui")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "enable_rtti_and_exceptions": [True, False],
+        "fPIC": [True, False],
+        "shared": [True, False],
+        "with_freetype_font_interface": [True, False],
+        "with_lua_bindings": [True, False],
+        "with_thirdparty_containers": [True, False]
+    }
+    default_options = {
+        "enable_rtti_and_exceptions": True,
+        "fPIC": True,
+        "shared": False,
+        "with_freetype_font_interface": True,
+        "with_lua_bindings": False,
+        "with_thirdparty_containers": True
+    }
+    generators = "cmake_find_package"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        tools.check_min_cppstd(self, self._minimum_cpp_standard)
+
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_freetype_font_interface:
+            self.requires("freetype/2.10.1")
+
+        if self.options.with_lua_bindings:
+            self.requires("lua/5.3.5")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        if not hasattr(self, "_cmake"):
+            self._cmake = CMake(self)
+
+            cmake_version_output = StringIO()
+            self.run("%s --version" % self._cmake._cmake_program, output=cmake_version_output)
+            cmake_version = Version(cmake_version_output.getvalue().split('\n', 1)[0].rsplit(' ', 1)[-1])
+
+            self._cmake.definitions["BUILD_LUA_BINDINGS"] = self.options.with_lua_bindings
+            self._cmake.definitions["BUILD_SAMPLES"] = False
+            self._cmake.definitions["DISABLE_RTTI_AND_EXCEPTIONS"] = not self.options.enable_rtti_and_exceptions
+            self._cmake.definitions["ENABLE_PRECOMPILED_HEADERS"] = cmake_version >= "3.16.0"
+            self._cmake.definitions["ENABLE_TRACY_PROFILING"] = False
+            self._cmake.definitions["NO_FONT_INTERFACE_DEFAULT"] = not self.options.with_freetype_font_interface
+            self._cmake.definitions["NO_THIRDPARTY_CONTAINERS"] = not self.options.with_thirdparty_containers
+
+            self._cmake.configure(source_folder=self._source_subfolder)
```

Comment:
You're not using a cmake wrapper script that calls `conan_basic_setup` from the `cmake` generator.
In conan v1, this is needed to apply correct compile options for e.g. Visual Studio.

e.g. this file for aws-c-common: https://github.com/conan-io/conan-center-index/blob/master/recipes/aws-c-common/all/CMakeLists.txt

You need to add it to `export_sources` + modify the arguments of `cmake.configure()`.

Reply:
It's all done. The only thing that remains is a question in my mind as to why exactly this is necessary. What compile options are being set? Does it have anything to do with the options MT/MD/MTd/MDd?

Reply:
Indeed, that's the culprit.